code/game/objects/items/holotape.dm

```diff
@@ -1,254 +0,0 @@
-			return
-		if(get_dist(start,end) >= MAX_TAPE_RANGE)
-			usr << "<span class='warning'>Your holotape segment is too long! It must be [MAX_TAPE_RANGE] tiles long or shorter!</span>"
-			return
-
-		var/turf/cur = start
-		var/dir
-		if(start.x == end.x)
-			var/d = end.y-start.y
-			if(d) d = d/abs(d)
-			end = get_turf(locate(end.x,end.y+d,end.z))
-			dir = "v"
-		else
-			var/d = end.x-start.x
-			if(d) d = d/abs(d)
-			end = get_turf(locate(end.x+d,end.y,end.z))
-			dir = "h"
-
-		var/can_place = 1
-		while (cur!=end && can_place)
-			if(cur.density == 1)
-				can_place = 0
-			else if(istype(cur, /turf/space))
-				can_place = 0
-			else
-				for(var/obj/O in cur)
-					if(!istype(O, /obj/item/holotape) && O.density)
-						can_place = 0
-						break
-			cur = get_step_towards(cur,end)
-		if(!can_place)
-			usr << "<span class='warning'>You can't project the [icon_base] holotape through that!</span>"
-			return
-
-		cur = start
-		var/tapetest = 0
-		while (cur!=end)
-			for(var/obj/item/holotape/Ptest in cur)
-				if(Ptest.icon_state == "[Ptest.icon_base]_[dir]")
-					tapetest = 1
-			if(tapetest != 1)
-				var/obj/item/holotape/P = new tape_type(cur)
-				P.icon_state = "[P.icon_base]_[dir]"
-			cur = get_step_towards(cur,end)
-
-		user.visible_message("[user] finishes projecting the length of [icon_base] holotape.", "<span class='notice'>You finish projecting the length of [icon_base] holotape.</span>")
-
-		charging = 1
-		spawn(40)
-			charging = 0
-
-/obj/item/tapeproj/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
-	if(charging)
-		return
-
-	if(proximity_flag == 0) // not adjacent
-		return
-
-	if(istype(target, /obj/machinery/door/airlock) || istype(target, /obj/machinery/door/firedoor) || istype(target, /obj/structure/window))
-		var/turf = get_turf(target)
-
-		if(locate(tape_type) in turf) //Don't you dare stack tape
-			return
-
-		if(istype(target, /obj/structure/window))
-			var/obj/structure/window/W = target
-			if(!(W.dir == 5) || !(W.fulltile == 1))
-				return
-
-		user << "<span class='notice'>You start projecting the [icon_base] holotape onto [target]...</span>"
-
-		if(!do_mob(user, target, 30))
-			return
-
-		var/atom/tape = new tape_type(turf)
-		tape.icon_state = "[icon_base]_door"
-		tape.layer = 3.2
-
-		user << "<span class='notice'>You project the [icon_base] holotape onto [target].</span>"
-
-		charging = 1
-		spawn(40)
-			charging = 0
-
-/obj/item/holotape/Bumped(mob/M)
-	if(!ismob(M))
-		return
-	if(iscarbon(M))
-		var/mob/living/carbon/C = M
-		if(C.m_intent == "walk")
-			var/turf/T = get_turf(src)
-			C.loc = T
-
-	if(M.has_unlimited_silicon_privilege)
-		var/turf/T = get_turf(src)
-		M.loc = T
-
-/obj/item/holotape/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
-	if(!density) return 1
-	if(air_group || (height==0)) return 1
-
-	if((mover.flags & PASSGLASS || istype(mover, /obj/effect/meteor) || mover.throwing == 1) )
```

Comment:
I'd bet anything there was some guy on Bay meticulously placing down holotape all round to block all meteors so they had to add this :P